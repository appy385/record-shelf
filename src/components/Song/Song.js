import React from 'react';
import PropTypes from 'prop-types';
import './Song.scss';
import heartGray from '../../assets/heart-gray.svg';
import heartRed from '../../assets/heart-red.svg';

const Song = ({ song, handleLike }) => (
  <div className="song-card" data-testid="song-card">
    <img className="song-image" src={song.albumArtUrl} alt="song" />
    <div className="song-container">
      <div>
        <div className="song-name">{song.name}</div>
        <div className="song-artist">{song.artist.name}</div>
      </div>
      <div className="song-image-container">
        {(song.like) ? <input type="image" className="song-heart-button" src={heartRed} alt="text" onClick={() => handleLike(song.id, !song.like)} />
          : <input type="image" className="song-heart-button" src={heartGray} alt="text" onClick={() => handleLike(song.id, !song.like)} />}
        <div className="song-likes">
          <span className="song-like-count">{song.count}</span>
        </div>
      </div>
    </div>
  </div>
);

Song.propTypes = {
  song: PropTypes.shape({
    id: PropTypes.string.isRequired,
    albumArtUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    artist: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    genre: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    publishedAt: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    like: PropTypes.bool.isRequired,
  }).isRequired,
  handleLike: PropTypes.func.isRequired,

};

export default Song;
