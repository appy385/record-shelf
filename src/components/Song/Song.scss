.song-card {
    width: calc((100% - (8 * 0.5rem + 8*1rem)) / 3);
    display: flex;
    flex-direction: column;
    margin: 0.5rem;
    padding: 1rem;
    border-radius: 0.25rem;
    background-color: #DFDFDF;
    box-shadow: 0 0.25rem 0.5rem 0 #9a9a9a;
}
.song-image{
    border-radius: 50%;
    width: 60%;
    margin: auto;
}
.song-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.song-name {
    font-weight: 400;
    font-size: 1rem;
    padding: 0.5rem 0 0 0.5rem;
}

.song-artist {
    font-weight: 700;
    font-size: 0.9rem;
    padding: 0 0 0 0.5rem;
}

.song-image-container {
    position: relative;
    display: inline-block;
}
.song-heart-button {

}

.song-like-count {
    position: absolute;

}

@media (max-width: 30rem) {
    .song-card{
        width: 100%;
    }
}
@media (max-width: 48rem) {
    .song-card {
        width:  35%
    }
}

