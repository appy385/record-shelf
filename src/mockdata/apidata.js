export const mockgetSongsApi = {
  data: {
    data: [
      {
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
        artist: { id: 'b22d7ed5-3ded-4958-a036-60453b7fc102', name: 'Neha Kakkar' },
        genre: { id: '37146752-2c95-4204-a845-a32f3cdce9c8', name: 'Bollywood' },
        id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
        name: 'Aankh Marey (From Simmba)',
        publishedAt: '2018-12-06T00:00:00',
      },
      {
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8',
        artist: { id: 'a9d57d30-182f-4297-b5d9-f1e765c86935', name: 'Phillip Phillips' },
        genre: { id: '8f63f8b8-a803-45da-8b2d-26a32ede9230', name: 'Country' },
        id: '08d7e81e-1c34-4a60-a70c-2ec11c8a6687',
        name: 'Gone, Gone, Gone',
        publishedAt: '2013-01-01T00:00:00',
      },

    ],
  },

};

export const newData = {};
