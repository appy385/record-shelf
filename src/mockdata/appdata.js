export const mockSongData = {
  albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8',
  artist: { id: 'a9d57d30-182f-4297-b5d9-f1e765c86935', name: 'Phillip Phillips' },
  genre: { id: '8f63f8b8-a803-45da-8b2d-26a32ede9230', name: 'Country' },
  id: '08d7e81e-1c34-4a60-a70c-2ec11c8a6687',
  name: 'Gone, Gone, Gone',
  publishedAt: '2013-01-01T00:00:00',
};

export const newS = {};
