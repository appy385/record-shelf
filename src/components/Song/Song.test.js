import React from 'react';
import { render, screen } from '@testing-library/react';
import Song from './Song';
import { mockSongData } from '../../mockdata/appdata';

describe(Song.name, () => {
  test('shoudld match snapshot', () => {
    const mockHandleLike = jest.fn();
    const { container } = render(<Song song={mockSongData} handleLike={mockHandleLike} />);
    expect(container).toMatchSnapshot();
  });
  test('shoudld have a gray heart image button', () => {
    const mockHandleLike = jest.fn();
    render(<Song song={mockSongData} handleLike={mockHandleLike} />);
    const element = screen.getByTestId('gray-heart');
    expect(element.tagName).toBe('INPUT');
  });
});
