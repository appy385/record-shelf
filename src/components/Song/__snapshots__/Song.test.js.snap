// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song shoudld match snapshot 1`] = `
<div>
  <div
    class="song-card"
    data-testid="song-card"
  >
    <img
      alt="song"
      class="song-image"
      src="https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8"
    />
    <div
      class="song-container"
    >
      <div>
        <div
          class="song-name"
        >
          Gone, Gone, Gone
        </div>
        <div
          class="song-artist"
        >
          Phillip Phillips
        </div>
      </div>
      <div
        class="song-image-container"
      >
        <input
          alt="text"
          class="song-heart-button"
          data-testid="gray-heart"
          src="heart-gray.svg"
          type="image"
        />
        <div
          class="song-likes"
        >
          <span
            class="song-like-count"
          >
            3
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
